arcgis-ios-sdk-samples/Utility network/Display content of utility network container/DisplayContentUtilityNetworkContainerViewController.swift

```diff
@@ -0,0 +1,296 @@
+                contentFeatures.forEach { content in
+                    let symbol = (content.featureTable as? AGSArcGISFeatureTable)?.layerInfo?.drawingInfo?.renderer?.symbol(for: content)
+                    overlay.graphics.add(AGSGraphic(geometry: content.geometry, symbol: symbol))
+                }
+                // The bounding box which defines the container view may be computed using the extent of the features it contains or centered around its geometry at the container's view scale.
+                var boundingBox: AGSGeometry?
+                if overlay.graphics.count == 1,
+                   let point = (overlay.graphics.firstObject as? AGSGraphic)?.geometry as? AGSPoint {
+                    self.mapView.setViewpointCenter(point, scale: containerElement.assetType.containerViewScale) { _ in
+                        guard let boundingBox = self.mapView.currentViewpoint(with: .boundingGeometry)?.targetGeometry else { return }
+                        self.getAssociationsWithExtent(boundingBox: boundingBox, overlay: overlay)
+                    }
+                } else {
+                    boundingBox = AGSGeometryEngine.bufferGeometry(overlay.extent, byDistance: 0.05)
+                    self.getAssociationsWithExtent(boundingBox: boundingBox!, overlay: overlay)
+                }
+            } else if let error = error {
+                self.presentAlert(error: error)
+            }
+        }
+    }
+    
+    /// Get associations for the specified extent and display its associations.
+    ///
+    /// - Parameters:
+    ///     - boundingBox: The gemeotry which represents the boundaries of the extent.
+    ///     - overlay: The overlay to add the symbols to.
+    func getAssociationsWithExtent(boundingBox: AGSGeometry, overlay: AGSGraphicsOverlay) {
+        // Add the bounding box symbol.
+        overlay.graphics.add(AGSGraphic(geometry: boundingBox, symbol: boundingBoxSymbol))
+        let geometry = AGSGeometryEngine.bufferGeometry(overlay.extent, byDistance: 0.05)!
+        mapView.setViewpointGeometry(geometry) { [weak self] _ in
+            guard let self = self else { return }
+            // Get the associations for this extent to display how content features are attached or connected.
+            self.utilityNetwork?.associations(withExtent: overlay.extent) { (containmentAssociations, error) in
+                containmentAssociations?.forEach { association in
+                    var symbol = AGSSymbol()
+                    if association.associationType == .attachment {
+                        symbol = self.attachmentSymbol
+                    } else {
+                        symbol = self.connectivitySymbol
+                    }
+                    overlay.graphics.add(AGSGraphic(geometry: association.geometry, symbol: symbol))
+                }
+                // Enable the bar button item to exit the container view.
+                self.exitBarButtonItem.isEnabled = true
```

Comment:
This should be moved up front to the beginning of the tap event. Otherwise there would be some timing bugs if you try to pan the map while get association call is still running.

Reply:
There is no tap event within this function. Do you mean move it out of this function and into geoView or beginning of `setViewpointGeometry`?

Reply:
Sorry for being vague. I commented on the wrong line... 😅  I mean to move the line below, `mapView.isUserInteractionEnabled = false` to the beginning of geoView -> didTapAtScreenPoint. So that once the user tap on a feature, they cannot pan or zoom the map, to get rid of timing bugs.